src/components/app/UrlManager.js

```diff
@@ -141,13 +141,24 @@ class UrlManager extends React.PureComponent<Props> {
       }
     }
 
+    // Profile sanitization and publishing can do weird things for the history API.
+    // Rather than write lots of complicated interactions, just prevent the back button
+    // from working when going between a published profile, and one that is not.
+    const isOneDataSourcePublic =
```

Comment:
Optional: I find this a bit hard to read with the variable names. What about disabling prettier for this if statement:

```js
    // prettier-ignore
    if (
      // Does the hash change?
      previousUrlState.hash !== newUrlState.hash ||
      (
        // Does the data source change?
        previousUrlState.dataSource !== newUrlState.dataSource &&

        // The data source changed, is one data source from the add-on?
        (previousUrlState.dataSource === 'from-addon' || newUrlState.dataSource === 'from-addon') &&

        // The data source changed, is one data source public?
        (previousUrlState.dataSource === 'public' || newUrlState.dataSource === 'public')
      )
    ) {
```